### DISCUSSION TITLE
Is it possible to do email verification and password/email changes without sending people to the dashboard?

### DISCUSSION THREAD

**Discussion by n-ham** [pocketbase/pocketbase#2771] [Q&A] [answered]
Is it possible with either Go or the Javascript API to verify email and handle password/email changes without having to direct users towards the PB dashboard URL in the emails? 

Example of how it'd work: 
1. User signs up and gets sent verification email
2. Email contains link back to website rather than PB dash
3. Website/client then communicates with PB dash to report the token user got in the email

That way the users would never be exposed to the PB dashboard, both lack of dark mode and just because it feels like a less authentic experience. 

Also is it possible to verify a password is correct when the user is already logged in? For example how GitHub will require people enter their password to delete a repository, can I confirm a user's password when they want to delete a project?

**Comment by ganigeorgiev** [comment, marked as answer]
You can change the default email confirmation url from the _"Admin UI > Settings > Mail settings > {template} > Action URL"_ input. The only required parameter is to have the `{TOKEN}` param in the url (either as path segment or query param).

In your custom page you have to read this token and send a confirm request to PocketBase with it. Examples and api info for the confirm apis can be found in the "API Preview" section for your auth collection.

**Reply by ganigeorgiev** [reply]
>  can I confirm a user's password when they want to delete a project?

There is no builtin functionality for this. You have to perform the check either in a middleware service or use PocketBase as framework and attach to the specific [event hook](https://pocketbase.io/docs/event-hooks/) to intercept the action. For verifying the user password you can use the [`record.ValidatePassword(password)`](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.16.5/models#Record.ValidatePassword).

**Reply by n-ham** [reply]
That'll work perfectly on all counts, thanks heaps!
